packages/flutter_tools/test/integration.shard/android_plugin_skip_unsupported_test.dart

```diff
@@ -0,0 +1,154 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:file_testing/file_testing.dart';
+import 'package:flutter_tools/src/base/file_system.dart';
+import 'package:flutter_tools/src/base/io.dart';
+import 'package:flutter_tools/src/cache.dart';
+
+import '../src/common.dart';
+import 'test_data/basic_project.dart';
+import 'test_data/legacy_settings_gradle_project.dart';
+import 'test_data/project.dart';
+import 'test_utils.dart';
+
+void main() {
+  late Directory tempDir;
+
+  setUp(() {
+    Cache.flutterRoot = getFlutterRoot();
+    tempDir = createResolvedTempDirectorySync('flutter_plugin_test.');
+  });
+
+  tearDown(() async {
+    tryToDelete(tempDir);
+  });
+
+  // Regression test for https://github.com/flutter/flutter/issues/97729 (#137115).
+  Future<ProcessResult> testPlugin({
+    required Project project,
+  }) async {
+    final String flutterBin = fileSystem.path.join(
+      getFlutterRoot(),
+      'bin',
+      'flutter',
+    );
+
+    // Create dummy plugin that supports iOS and Android.
+    processManager.runSync(<String>[
+      flutterBin,
+      ...getLocalEngineArguments(),
+      'create',
+      '--template=plugin',
+      '--platforms=ios,android',
+      'test_plugin',
+    ], workingDirectory: tempDir.path);
+
+    final Directory pluginAppDir = tempDir.childDirectory('test_plugin');
+
+    // Override pubspec to drop support for the Android implementation.
+    final File pubspecFile = pluginAppDir.childFile('pubspec.yaml');
+    const String pubspecYamlSrc = r'''
```

Comment:
I'm worried about unrelated upstream changes to the pubspec.yaml being lost here.

Could you express this as:

1) open pubspec.yaml that flutter create made.
2) process all the lines in (1) replacing the ones you don't like with a comment prefix
3) update pubspec.yaml

This will make the test less sensitive to upstream changes in pubspec.yaml files

Reply:
This would also help the reader understand what you're changing in the file.

Reply:
Done.